### PR TITLE
arch: riscv: some IRQ cleanups

### PR DESCRIPTION
**Pull request by gmarull**

### CODE REVIEW

**Review by fabiobaltieri — DISMISSED** [review]

File: arch/riscv/Kconfig [hunk 1]
```diff
@@ -172,6 +172,13 @@ config RISCV_HAS_CLIC
 	help
 	  Does the SOC provide support for a Core-Local Interrupt Controller (CLIC).
 
+config RISCV_SOC_ISR_CHECK
```
Comment (carlocaione) [inline arch/riscv/Kconfig:175]: not to bikeshedding but this config name could be better. We are checking whether the exception is caused by an IRQ (vs EXC). This has nothing to do with ISR really.

Comment (gmarull) [reply arch/riscv/Kconfig:175]: good point. also investigating CI failures now.

**Comment by npitre** [comment]
You do:

```
        srli t0, t0, RISCV_MCAUSE_IRQ_BIT
```

This is wrong as `RISCV_MCAUSE_IRQ_BIT` is defined as the actual bit not the
bit number usable as a shift.

**Comment by gmarull** [comment]
> You do:
> 
> ```
>         srli t0, t0, RISCV_MCAUSE_IRQ_BIT
> ```
> 
> This is wrong as `RISCV_MCAUSE_IRQ_BIT` is defined as the actual bit not the bit number usable as a shift.

my bad, I missed that this definition holds the mask. fixed.

File: arch/riscv/Kconfig (hunk 1 above)
Comment (gmarull) [reply arch/riscv/Kconfig]: renamed to `CONFIG_RISCV_SOC_EXCEPTION_FROM_IRQ`

**Review by npitre — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Comment by carlescufi** [comment]
@fkokosinski could you take a look?

**Review by fkokosinski — APPROVED** [review]
LGTM

**Pull request merged**